### DISCUSSION TITLE
TeleportRandom Killing Server Process

### DISCUSSION THREAD

**Discussion by MoeXIV** [EssentialsX/Essentials#4840] [Q&A] [answered]
I wish I had more information to give, but truly this is to the extent I can find.

When running /tpr the server gets killed. Nothing in logs, not enough time to paste a timings. Not even a thread dump, the server just ramps up to like 300% CPU within a second or two and then the process gets killed.

Are there any other troubleshooting steps we can do? 
Here's our Essentials server dump: https://essentialsx.net/dump.html?id=c5b99f74f5514ad39d6b70e2da19844a

**Comment by osfanbuff63** [comment]
I suspect that since /tpr teleports you 1,000,000 blocks away (ish), that the server is getting overloaded. The only possible way to fix this (I think) is change /tpr's distance to not so far away, although I'm not sure if that would fix it or not.

**Comment by pop4959** [comment, marked as answer]
If you don't see any error / thread dump and the process is outright killed it could be that the server is running out of memory, or something of the sort, which could easily just be due to chunks generating when teleported. Try doing what @osfanbuff63 mentioned and reduce the teleport radius (with `/settpr`) to something that wouldn't generate new chunks in the world.

**Reply by MoeXIV** [reply]
Turned out to be that the "world" world was weirdly almost entirely different oceans inside the max range radius (which are listed as restricted biomes in TPR), so I imagine it was searching in a bunch of different locations and loading an insane amount of chunks. Max range was only set to 10000 which I felt like wasn't enough to cause a crash.

Regenned the world, the terrain looks pretty normal right now, and TPR worked as intended although still using a lot of CPU. May just have to gen the entire world before release.

I appreciate the help, you guys!

P.S: server was running on 7GB RAM on a decent (but not great) VPS.
